src/main/java/uk/gov/ons/ssdc/caseprocessor/utils/RedactHelper.java

```diff
@@ -0,0 +1,56 @@
+package uk.gov.ons.ssdc.caseprocessor.utils;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.Map;
+
+public class RedactHelper {
+  private static final String REDACTION_FAILURE = "Failed to redact sensitive data";
+  private static final String REDACTION_TARGET = "setSampleSensitive";
+  private static final ObjectMapper objectMapper = ObjectMapperFactory.objectMapper();
+
+  public static Object redact(Object rootObjectToRedact) {
+    try {
+      Object rootObjectToRedactDeepCopy =
+          objectMapper.readValue(
+              objectMapper.writeValueAsString(rootObjectToRedact), rootObjectToRedact.getClass());
+      recursivelyRedact(
+          rootObjectToRedactDeepCopy,
+          REDACTION_TARGET,
+          rootObjectToRedactDeepCopy.getClass().getPackageName());
+      return rootObjectToRedactDeepCopy;
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException(REDACTION_FAILURE, e);
+    }
+  }
+
+  private static void recursivelyRedact(Object object, String methodToFind, String packageName) {
+    Arrays.stream(object.getClass().getMethods())
+        .filter(item -> Modifier.isPublic(item.getModifiers()))
+        .forEach(
+            method -> {
+              if (method.getName().startsWith("get")
+                  && method.getReturnType().getPackageName().equals(packageName)) {
+                try {
+                  Object invokeResult = method.invoke(object);
+                  if (invokeResult != null) {
+                    recursivelyRedact(invokeResult, methodToFind, packageName);
+                  }
+                } catch (IllegalAccessException | InvocationTargetException e) {
+                  throw new RuntimeException(REDACTION_FAILURE, e);
+                }
+              }
+
+              if (method.getName().equals(methodToFind)) {
+                try {
+                  method.invoke(object, Map.of("REDACTED", "REDACTED"));
```

Comment:
Do we want to redact the key too? Maybe helpful if the logs at least keeps the record of which data was changed.

Reply:
It's gonna be a lot more tricky to redact just the values. Not impossible, but tricky.

What the code does, for now, is to replace the entire map with a single entry: "REDACTED" (key) -> "REDACTED" (value)